metadata-writer/rockcraft.yaml

```
@@ -34,28 +34,28 @@ parts:
     plugin: python
     source: https://github.com/kubeflow/pipelines.git
     source-subdir: backend/metadata_writer
-    source-tag: 2.3.0
+    source-tag: 2.4.0
     stage-packages:
       # sync this python version to the base image in the Dockerfile.
       # Also keep this in sync with PARTS_PYTHON_INTERPRETER below.
-    - python3.8-venv
+    - python3.9-venv
     python-requirements:
     - requirements.txt
     build-environment: 
-    - PARTS_PYTHON_INTERPRETER: python3.8
+    - PARTS_PYTHON_INTERPRETER: python3.9
 
   python3-is-python3.8:
```

We could probably rename this
```suggestion
  python3-is-python3.9:
```

---

Good catch